Updates 'Getting ready for production' page (#113679)
elastic/elasticsearch: docs/reference/intro.asciidoc

| Use KQL to filter documents where a value for a field exists, matches a given value, or is within a given range.
| N/A

|===

// New html page
[[scalability]]
=== Getting ready for production

Many teams rely on {es} to run their key services. To keep these services running, you can design your {es} deployment
to keep {es} available, even in case of large-scale outages. To keep it running fast, you also can design your
deployment to be responsive to production workloads.

{es} is built to be always available and to scale with your needs. It does this using a distributed architecture.
By distributing your cluster, you can keep Elastic online and responsive to requests.

In case of failure, {es} also offers tools for cross-cluster replication and cluster snapshots that can
help you fall back or recover quickly. You can also use cross-cluster replication to serve requests based on the
geographic location of your users and your resources.

{es} also offers security and monitoring tools to help you keep your cluster highly available.

[discrete]
[[use-multiple-nodes-shards]]

> ==== Use multiple nodes and shards for better performance and resilience

Reviewer: ```suggestion
==== Use multiple nodes and shards for better performance and resilience
++++
<titleabbrev>Use multiple nodes and shards</titleabbrev>
++++
```

abbreviation is nicer in the "On this page" section

[NOTE]
====
Nodes and shards are what make {es} distributed and scalable.

These concepts aren’t essential if you’re just getting started. How you <<elasticsearch-intro-deploy,deploy {es}>> in production determines what you need to know:

* *Self-managed {es}*: You are responsible for setting up and managing nodes, clusters, shards, and replicas. This includes
managing the underlying infrastructure, scaling, and ensuring high availability through failover and backup strategies.
* *Elastic Cloud*: Elastic can autoscale resources in response to workload changes. Choose from different deployment types
to apply sensible defaults for your use case. A basic understanding of nodes, shards, and replicas is still important.
* *Elastic Cloud Serverless*: You don’t need to worry about nodes, shards, or replicas. These resources are 100% automated
on the serverless platform, which is designed to scale with your workload.
====

> You can add servers _(nodes)_ to a cluster to increase capacity, and {es} automatically distributes your data and query load

Reviewer: ```
  (\_nodes_/)
   \  ^_^  /
    \(___)/
```

of one or more physical shards, where each shard is a self-contained Lucene index containing a subset of the documents in
the index. By distributing the documents in an index across multiple shards, and distributing those shards across multiple
nodes, {es} increases indexing and query capacity.

There are two types of shards: _primaries_ and _replicas_. Each document in an index belongs to one primary shard. A replica
shard is a copy of a primary shard. Replicas maintain redundant copies of your data across the nodes in your cluster. 
This protects against hardware failure and increases capacity to serve read requests like searching or retrieving a document.

[TIP]
====
The number of primary shards in an index is fixed at the time that an index is created, but the number of replica shards can
be changed at any time, without interrupting indexing or query operations.
====

Shard copies in your cluster are automatically balanced across nodes to provide scale and high availability. All nodes are
aware of all the other nodes in the cluster and can forward client requests to the appropriate node. This allows {es}
to distribute indexing and query load across the cluster.

> If you’re exploring {es} for the first time, then you can use a cluster with a single node and create indices

Reviewer 2: ```suggestion
If you’re exploring {es} for the first time or working in a development environment, then you can use a cluster with a single node and create indices
```

with multiple shards to increase performance and resilience.

// TODO - diagram

To learn about optimizing the number and size of shards in your cluster, refer to <<size-your-shards,Size your shards>>. 
To learn about how read and write operations are replicated across shards and shard copies, refer to <<docs-replication,Reading and writing documents>>.
To adjust how shards are allocated and balanced across nodes, refer to <<shard-allocation-relocation-recovery,Shard allocation, relocation, and recovery>>.

[discrete]
[[ccr-disaster-recovery-geo-proximity]]
==== CCR for disaster recovery and geo-proximity

To effectively distribute read and write operations across nodes, the nodes in a cluster need good, reliable connections
to each other. To provide better connections, you typically co-locate the nodes in the same data center or nearby data centers.

Co-locating nodes in a single location exposes you to the risk of a single outage taking your entire cluster offline. To
maintain high availability, you can prepare a second cluster that can take over in case of disaster by implementing

> ross-cluster replication (CCR).

Reviewer 2: ```suggestion
cross-cluster replication (CCR).
```

maintain high availability, you can prepare a second cluster that can take over in case of disaster by implementing
ross-cluster replication (CCR).

As a starting point:
CCR provides a way to automatically synchronize indices from your primary cluster to a secondary remote cluster that
can serve as a hot backup. If the primary cluster fails, the secondary cluster can take over.

* Aim to keep the average shard size between a few GB and a few tens of GB. For
  use cases with time-based data, it is common to see shards in the 20GB to 40GB
  range.
You can also use CCR to create secondary clusters to serve read requests in geo-proximity to your users.

* Avoid the gazillion shards problem. The number of shards a node can hold is
  proportional to the available heap space. As a general rule, the number of
  shards per GB of heap space should be less than 20.
Learn more about <<xpack-ccr,cross-cluster replication>> and about <<high-availability-cluster-design,designing for resilience>>.

The best way to determine the optimal configuration for your use case is
through https://www.elastic.co/elasticon/conf/2016/sf/quantitative-cluster-sizing[

> testing with your own data and queries].

Reviewer 2: Bogi, can you make sure that the concepts deleted here are represented in [size your shards](https://www.elastic.co/guide/en/elasticsearch/reference/current/size-your-shards.html)? just worried about removing basic guidance

Reviewer: FYI I haven't found any information loss

Author: I went through everything and it looks to me like nothing important is missing from the 'Size your shards' chapter.

> You can also take <<snapshot-restore,snapshots>> of your cluster that can be restored in case of failure.

Reviewer 2: @leemthompo do you think this mention is enough since we introduced it in the overview as well? Should there be another section for snapshot/restore? leaning toward yes

Reviewer: Not sure really, the page is nice and digestible right now

> monitoring, and administrative features that are integrated into {es} enable you to use https://www.elastic.co/guide/en/kibana/8.15/introduction.html[Kibana] as a

Reviewer: ```suggestion
monitoring, and administrative features that are integrated into {es} enable you to use {kibana-ref}/introduction.html[Kibana] as a
```

| {kib} Query Language (KQL) is a text-based query language for filtering data when you access it through the {kib} UI.
| Use KQL to filter documents where a value for a field exists, matches a given value, or is within a given range.
| N/A

|===

// New html page
[[scalability]]
=== Get ready for production

Many teams rely on {es} to run their key services. To keep these services running, you can design your {es} deployment
to keep {es} available, even in case of large-scale outages. To keep it running fast, you also can design your
deployment to be responsive to production workloads.

{es} is built to be always available and to scale with your needs. It does this using a distributed architecture.
By distributing your cluster, you can keep Elastic online and responsive to requests.

In case of failure, {es} offers tools for cross-cluster replication and cluster snapshots that can
help you fall back or recover quickly. You can also use cross-cluster replication to serve requests based on the
geographic location of your users and your resources.

{es} also offers security and monitoring tools to help you keep your cluster highly available.

[discrete]
[[use-multiple-nodes-shards]]

> ==== Use multiple nodes and shards for better performance and resilience
> ++++
> <titleabbrev>Use multiple nodes and shards</titleabbrev>
> ++++

Reviewer: ```suggestion
==== Use multiple nodes and shards
```

turns out the abbreviation is messing up the left-hand nav
